### ISSUE TITLE
Beautifying Gherkin throws error "TypeError: require(...).Lexer is not a function"

### ISSUE THREAD

**Issue by hun1ancer** [Glavin001/atom-beautify#1293]
TypeError: require(...).Lexer is not a function
    at Object.<anonymous> (file:///C:/Users/Wtfy~chaosir/.atom/packages/atom-beautify-master/src/beautifiers/gherkin.coffee:6:28)
    at Object.<anonymous> (file:///C:/Users/Wtfy~chaosir/.atom/packages/atom-beautify-master/src/beautifiers/gherkin.coffee:1:1)
    at Module._compile (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .coffee](C:UsersWtfy~chaosirAppDataLocalatomapp-1.11.2resourcesapp.asarsrccompile-cache.js:208:21)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\src\native-compile-cache.js:50:27)
    at file:///C:/Users/Wtfy~chaosir/.atom/packages/atom-beautify-master/src/beautifiers/index.coffee:104:20
    at arrayMap (C:\Users\Wtfy~chaosir.atom\packages\node_modules\lodash\lodash.js:645:23)
    at Function.map (C:\Users\Wtfy~chaosir.atom\packages\node_modules\lodash\lodash.js:9505:14)
    at new Beautifiers (file:///C:/Users/Wtfy~chaosir/.atom/packages/atom-beautify-master/src/beautifiers/index.coffee:103:22)
    at Object.<anonymous> (file:///C:/Users/Wtfy~chaosir/.atom/packages/atom-beautify-master/src/beautify.coffee:10:18)
    at Object.<anonymous> (file:///C:/Users/Wtfy~chaosir/.atom/packages/atom-beautify-master/src/beautify.coffee:2:1)
    at Module._compile (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .coffee](C:UsersWtfy~chaosirAppDataLocalatomapp-1.11.2resourcesapp.asarsrccompile-cache.js:208:21)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\src\native-compile-cache.js:50:27)
    at Package.module.exports.Package.requireMainModule (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\src\package.js:721:27)
    at Package.module.exports.Package.activateNow (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\src\package.js:173:16)
    at C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\src\package.js:801:25
    at Function.module.exports.Emitter.simpleDispatch (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\node_modules\event-kit\lib\emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\node_modules\event-kit\lib\emitter.js:129:28)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\src\command-registry.js:241:20)
    at CommandRegistry.handleCommandEvent (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\src\command-registry.js:3:61)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\src\command-registry.js:160:19)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchContextMenuCommand (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\src\atom-environment.js:1133:28)
    at EventEmitter.outerCallback (C:\Users\Wtfy~chaosir\AppData\Local\atom\app-1.11.2\resources\app.asar\src\application-delegate.js:354:25)
    at emitThree (events.js:110:13)
    at EventEmitter.emit (events.js:188:7)

**Comment by Evanxuke** [comment]
Hi @hun1ancer,What is your problem solved?

**Comment by prettydiff** [comment]
The appears to be on this line: https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/gherkin.coffee#L6

The likely problem is that the gherkin package failed to download and so the require function returns undefined.  Executing a method off undefined will always throw an error.  I recommend downloading Atom Beautify again to see if this solves the problem.

**Comment by Glavin001** [comment]
Let us know if you need more help. Issues like this are often problems with `apm` installing and reinstalling Atom-Beautify can resolve them.

**Comment by hyunchi** [comment]
when i git clone 
TypeError: require(...).Lexer is not a function

**Comment by prettydiff** [comment]
@hyunchi try reinstalling  `apm install atom-beautify`

**Comment by hyunchi** [comment]
But net fast very slow in china,when I use apm install,

**Comment by prettydiff** [comment]
If it is very slow in China I suspect that might explain a huge number of issues we see with missing dependencies.

**Comment by hyunchi** [comment]
And so you are,apm cannot be used probably Chinese Government's Network firewall

**Comment by prettydiff** [comment]
Keeping everybody informed I have fixed biddle.  Check it out now - https://github.com/prettydiff/biddle

I will republish Pretty Diff using biddle in production tomorrow.  I am not ready to give it to Atom Beautify just yet though.  I actually need users to play around with this thing for a bit so that I can get a sense of where the pain points are and just how bad it works for fails for other people.  I wrote a getting started document at https://github.com/prettydiff/biddle/blob/master/documentation/gettingstarted.md

The documentation might be easier to read from the command line though as I included a markdown to CLI parser in biddle.  The number at the end defines the word wrap limit:

    node biddle markdown documentation/gettingstarted.md 80
